### PR TITLE
[JENKINS-69229] Revert `trilead-api` update

### PR DESCRIPTION
**Pull request by jglick**
Reverting #1374. https://github.com/jenkinsci/bom/pull/1378#issuecomment-1205330558

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by jtnord — APPROVED** [review]

**Review by jetersen — APPROVED** [review]

**Comment by jglick** [comment]
OK let us go with this one for now.

**Review by MarkEWaite — APPROVED** [review]

**Review by basil — APPROVED** [review]

**Comment by jglick** [comment]
https://github.com/jenkinsci/bom/pull/1379/checks?check_run_id=7686535482 (second flake in this PR)

```
WARNING	o.j.p.w.cps.CpsFlowExecution#cleanUpHeap: failed to clean up memory from Owner[demo/1:demo #1]
java.lang.ClassCastException: class java.io.ObjectStreamClass$Caches$1 cannot be cast to class java.util.concurrent.ConcurrentMap (java.io.ObjectStreamClass$Caches$1 and java.util.concurrent.ConcurrentMap are in module java.base of loader 'bootstrap')
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.cleanUpObjectStreamClassCaches(CpsFlowExecution.java:1426)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.cleanUpLoader(CpsFlowExecution.java:1321)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.cleanUpLoader(CpsFlowExecution.java:1300)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.cleanUpLoader(CpsFlowExecution.java:1310)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.cleanUpHeap(CpsFlowExecution.java:1287)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:447)
	at …
```

File: .github/dependabot.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ updates:
       - "jglick"
     schedule:
       interval: "daily"
+    ignore:
+      # TODO https://github.com/jenkinsci/bom/pull/1374#issuecomment-1205302760 pending new version
+      - dependency-name: trilead-api
```
Comment (jglick) [inline .github/dependabot.yml:21]: https://github.com/jenkinsci/bom/pull/1382#issuecomment-1207121920

Comment (jglick) [reply .github/dependabot.yml:21]: a86fa42

Comment (jglick) [reply .github/dependabot.yml:21]: https://github.com/jenkinsci/trilead-api-plugin/releases/tag/2.72.v2a_3236754f73 includes https://github.com/jenkinsci/trilead-api-plugin/pull/55 which I guess is now safe for newer lines @kuisathaverat?

Comment (kuisathaverat) [reply .github/dependabot.yml:21]: yes, it is safe for 2.261+

**Pull request merged**